django/db/models/sql/datastructures.py

```diff
@@ -114,17 +114,28 @@ def relabeled_clone(self, change_map):
             self.join_field, self.nullable, filtered_relation=filtered_relation,
         )
 
-    def equals(self, other, with_filtered_relation):
+    @property
+    def identity(self):
         return (
-            isinstance(other, self.__class__) and
-            self.table_name == other.table_name and
-            self.parent_alias == other.parent_alias and
-            self.join_field == other.join_field and
-            (not with_filtered_relation or self.filtered_relation == other.filtered_relation)
+            self.__class__,
```

Comment:
And is this required? (Particularly if we use `self.__class__` in `__eq__()`.)
```suggestion
```
It feels like this is a hangover from the original not having the `return NotImplemented` branch.

Reply:
So `.equals()` will now use the new `.identity` property, but the question is whether this is still required. If there is no chance that `.equals()` will compare objects of different types, it is unnecessary.